internal/command/volumes/volumes.go

```diff
@@ -80,3 +86,82 @@ func countVolumesMatchingName(ctx context.Context, volumeName string) (int32, er
 
 	return matches, nil
 }
+
+func renderTable(ctx context.Context, volumes []api.Volume, app *api.App, out io.Writer) error {
+	apiClient := client.FromContext(ctx).API()
+	rows := make([][]string, 0, len(volumes))
+	for _, volume := range volumes {
+		var attachedVMID string
+
+		if app.PlatformVersion == "machines" {
+			if volume.AttachedMachine != nil {
+				attachedVMID = *volume.AttachedMachine
+			}
+		} else {
+			names, err := apiClient.GetAllocationTaskNames(ctx, app.Name)
+			if err != nil {
+				return err
+			}
+
+			if volume.AttachedAllocation != nil {
+				attachedVMID = *volume.AttachedAllocation
+
+				taskName, ok := names[*volume.AttachedAllocation]
+
+				if ok && taskName != "app" {
+					attachedVMID = fmt.Sprintf("%s (%s)", *volume.AttachedAllocation, taskName)
+				}
+			}
+		}
+
+		rows = append(rows, []string{
+			volume.ID,
+			volume.State,
+			volume.Name,
+			strconv.Itoa(volume.SizeGb) + "GB",
+			volume.Region,
+			volume.Zone,
+			fmt.Sprint(volume.Encrypted),
+			attachedVMID,
+			humanize.Time(volume.CreatedAt),
+		})
+	}
+
+	return render.Table(out, "", rows, "ID", "State", "Name", "Size", "Region", "Zone", "Encrypted", "Attached VM", "Created At")
+}
+
+func selectVolume(ctx context.Context, volumes []api.Volume, app *api.App) (*api.Volume, error) {
+	if len(volumes) == 0 {
+		return nil, fmt.Errorf("no volumes found in app '%s'", app.Name)
+	}
+	out := new(bytes.Buffer)
+	err := renderTable(ctx, volumes, app, out)
+	if err != nil {
+		return nil, err
+	}
+	volumeLines := make([]string, 0)
+	scanner := bufio.NewScanner(out)
+	title := ""
+	for scanner.Scan() {
+		text := scanner.Text()
+		if text == "" {
+			continue
+		}
+		if title == "" {
+			title = text
+			continue
+		}
+		volumeLines = append(volumeLines, text)
+	}
+	selected := 0
+	fmt.Println(title)
+	prompt := &survey.Select{
+		Message:  "Select volume:",
+		Options:  volumeLines,
+		PageSize: 15,
+	}
+	if err := survey.AskOne(prompt, &selected); err != nil {
+		return nil, fmt.Errorf("selecting volume: %w", err)
+	}
```

Comment:
Perhaps use the `Select` function from `github.com/superfly/flyctl/internal/prompt`? It's ultimately just a wrapper over `github.com/AlecAivazis/survey/v2`, but it provides some tweaks (like overriding the color scheme).

Reply:
This is a good idea. I don't remember if that helper function correctly handles non-interactive prompts, but it'd be nice to make sure corner cases like that are handled in one place so we're not repeating tedious code everywhere.

Reply:
Yes, it also looks like this wrapper helps return an error when running non-interactively as well which is important.